src/nwb_benchmarks/core/_streaming.py

```diff
@@ -179,3 +181,39 @@ def read_hdf5_nwbfile_ros3(s3_url: str, retry: bool = True) -> Tuple[pynwb.NWBFi
         retries = None
         nwbfile = io.read()
     return (nwbfile, io, retries)
+
+
+def read_zarr(s3_url: str, force_no_consolidated_metadata=False) -> zarr.Group:
+    """
+    Open a Zarr file from an S3 URL using the built-in fsspec support in Zarr.
+
+    Returns
+    -------
+    file : zarr.Group
+       The zarr.Group object representing the opened file
+    """
+    if force_no_consolidated_metadata:
+        zarrfile = zarr.open(store=s3_url, mode="r", storage_options=dict(anon=True))
+    else:
+        try:
+            zarrfile = zarr.open_consolidated(store=s3_url, mode="r", storage_options=dict(anon=True))
+        except KeyError:
+            zarrfile = zarr.open(store=s3_url, mode="r", storage_options=dict(anon=True))
+    return zarrfile
+
+
+def read_zarr_nwbfile(s3_url: str, force_no_consolidated_metadata=False) -> Tuple[pynwb.NWBFile, hdmf_zarr.NWBZarrIO]:
```

Comment:
Here it might be clearer to just pass the `mode` and add the details of the `r-` meaning into the docstring?

Reply:
If we want to avoid the try/accept logic for checking if consolidated metadata is available, then we should open the file ourselves in `read_zarr` , because otherwise we run into the same issue in `hdmf_zarr` where that logic actually makes sense, because we don't want the read to just fail there. I'll update the PR and then ping you so you can take a look.

Reply:
Sounds good 👍